addons/tls13-middlebox/bootstrap.js

```diff
@@ -194,6 +175,41 @@ function hasUserSetPreference() {
     return false;
 }
 
+// extracting the root certificate from the chain
+function extractRootCert(cert) {
+    let root_cert = cert;
+
+    while (getFieldValue(root_cert, 'issuer')) {
+        root_cert = getFieldValue(root_cert, 'issuer');
+    }
+
+    return root_cert;
+}
+
+// returns true if there is at least one non-builtin root certificate is installed
+function isNonBuiltInRootCertInstalled() {
+    let certDB = Cc["@mozilla.org/security/x509certdb;1"].getService(Ci.nsIX509CertDB);
+
+    var iter = certDB.getCerts().getEnumerator();
+
+    while (iter.hasMoreElements()) {
+        let cert = iter.getNext().QueryInterface(Ci.nsIX509Cert);
+
+        // extract the root certificate for the current certificate
+        let root_cert = extractRootCert(cert);
```

Comment:
This seems to do something different. It goes over all the certs and finds the root they chain to.